pkg/config/settings/http/client.go

```diff
@@ -30,35 +30,46 @@ func NewClient(c *http.Client, baseURL string, targetProcessName string, clientO
 	return &runtimeSettingsHTTPClient{c, baseURL, targetProcessName, clientOptions}
 }
 
-func (rc *runtimeSettingsHTTPClient) FullConfig() (string, error) {
-	r, err := util.DoGet(rc.c, rc.baseURL, rc.clientOptions.CloseConnection)
+func (rc *runtimeSettingsHTTPClient) doGet(url string, formatError bool) (string, error) {
+	r, err := util.DoGet(rc.c, url, rc.clientOptions.CloseConnection)
 	if err != nil {
-		var errMap = make(map[string]string)
+		errMap := make(map[string]string)
 		_ = json.Unmarshal(r, &errMap)
 		// If the error has been marshalled into a json object, check it and return it properly
 		if e, found := errMap["error"]; found {
 			return "", fmt.Errorf(e)
 		}
+		if formatError {
+			return "", fmt.Errorf("Could not reach %s: %v \nMake sure the %s is running before requesting the runtime configuration and contact support if you continue having issues", rc.targetProcessName, err, rc.targetProcessName)
+		}
+		return "", err
+	}
+	return string(r), nil
+}
 
-		return "", fmt.Errorf("Could not reach %s: %v \nMake sure the %s is running before requesting the runtime configuration and contact support if you continue having issues", rc.targetProcessName, err, rc.targetProcessName)
+func (rc *runtimeSettingsHTTPClient) FullConfig() (string, error) {
+	r, err := rc.doGet(rc.baseURL, true)
+	if err != nil {
+		return "", err
 	}
+	return string(r), nil
+}
 
+func (rc *runtimeSettingsHTTPClient) FullConfigBySource() (string, error) {
+	r, err := rc.doGet(fmt.Sprintf("%s/%s", rc.baseURL, "by-source"), true)
+	if err != nil {
+		return "", err
+	}
 	return string(r), nil
 }
 
 func (rc *runtimeSettingsHTTPClient) List() (map[string]settingsComponent.RuntimeSettingResponse, error) {
-	r, err := util.DoGet(rc.c, fmt.Sprintf("%s/%s", rc.baseURL, "list-runtime"), rc.clientOptions.CloseConnection)
+	r, err := rc.doGet(fmt.Sprintf("%s/%s", rc.baseURL, "list-runtime"), false)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Same as above
```suggestion
	r, err := rc.doGet(fmt.Sprintf("%s/list-runtime", rc.baseURL), false)
```